lib/transmuxer/mpeg_audio.js

```diff
@@ -0,0 +1,190 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.transmuxer.MpegAudio');
+
+
+/**
+ *  MPEG parser utils
+ */
+shaka.transmuxer.MpegAudio = class {
+  /**
+   * @param {!Uint8Array} data
+   * @param {!number} offset
+   * @return {?{sampleRate: number, channelCount: number,
+   * frameLength: number, samplesPerFrame: number}}
+   */
+  static parseHeader(data, offset) {
+    const MpegAudio = shaka.transmuxer.MpegAudio;
+
+    const mpegVersion = (data[offset + 1] >> 3) & 3;
+    const mpegLayer = (data[offset + 1] >> 1) & 3;
+    const bitRateIndex = (data[offset + 2] >> 4) & 15;
+    const sampleRateIndex = (data[offset + 2] >> 2) & 3;
+    if (mpegVersion !== 1 && bitRateIndex !== 0 &&
+      bitRateIndex !== 15 && sampleRateIndex !== 3) {
+      const paddingBit = (data[offset + 2] >> 1) & 1;
+      const channelMode = data[offset + 3] >> 6;
+      const columnInBitrates =
+        mpegVersion === 3 ? 3 - mpegLayer : mpegLayer === 3 ? 3 : 4;
+      const bitRate = MpegAudio.BITRATES_MAP_[
+          columnInBitrates * 14 + bitRateIndex - 1] * 1000;
+      const columnInSampleRates =
+        mpegVersion === 3 ? 0 : mpegVersion === 2 ? 1 : 2;
+      const sampleRate = MpegAudio.SAMPLINGRATE_MAP_[
+          columnInSampleRates * 3 + sampleRateIndex];
+      // If bits of channel mode are `11` then it is a single channel (Mono)
+      const channelCount = channelMode === 3 ? 1 : 2;
+      const sampleCoefficient =
+          MpegAudio.SAMPLES_COEFFICIENTS_[mpegVersion][mpegLayer];
+      const bytesInSlot = MpegAudio.BYTES_IN_SLOT_[mpegLayer];
+      const samplesPerFrame = sampleCoefficient * 8 * bytesInSlot;
+      const frameLength =
+        Math.floor((sampleCoefficient * bitRate) / sampleRate + paddingBit) *
+        bytesInSlot;
+
+      const userAgent = navigator.userAgent || '';
```

Comment:
We have utilities for detecting browsers and versions in shaka.util.Platform

Reply:
Yes, but the utility is only for chrome browser, this is for all browser with Chrome string.